app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -510,25 +518,29 @@ bool OpenXRInputSource::GetHandTrackingInfo(const XrFrameState& frameState, XrSp
     if (OpenXRExtensions::sXrLocateHandJointsEXT == XR_NULL_HANDLE || mHandTracker == XR_NULL_HANDLE)
         return false;
 
-    mHasHandJoints = false;
-    mHasAimState = false;
-
     // Update hand locations
     XrHandJointsLocateInfoEXT locateInfo { XR_TYPE_HAND_JOINTS_LOCATE_INFO_EXT };
     locateInfo.baseSpace = localSpace;
     locateInfo.time = frameState.predictedDisplayTime;
 
-    mAimState.status = 0;
-    mAimState.next = XR_NULL_HANDLE;
-
+    XrHandTrackingAimStateFB aimState { XR_TYPE_HAND_TRACKING_AIM_STATE_FB, XR_NULL_HANDLE, 0  };
     XrHandJointLocationsEXT jointLocations { XR_TYPE_HAND_JOINT_LOCATIONS_EXT };
     jointLocations.jointCount = XR_HAND_JOINT_COUNT_EXT;
     jointLocations.jointLocations = mHandJoints.data();
-    jointLocations.next = &mAimState;
+    jointLocations.next = &aimState;
 
     CHECK_XRCMD(OpenXRExtensions::sXrLocateHandJointsEXT(mHandTracker, &locateInfo, &jointLocations));
     mHasHandJoints = jointLocations.isActive;
-    mHasAimState = (mAimState.status & XR_HAND_TRACKING_AIM_VALID_BIT_FB) != 0;
+    if (mSupportsFBHandTrackingAim) {
+        mHasAimState = aimState.status & XR_HAND_TRACKING_AIM_VALID_BIT_FB;
+        if (mHasAimState)
+            mHandAimPose = aimState.aimPose;
+    } else {
+        auto aimJoint = jointLocations.jointLocations[XR_HAND_JOINT_PALM_EXT];
+        mHasAimState = aimJoint.locationFlags & XR_SPACE_LOCATION_ORIENTATION_TRACKED_BIT;
+        if (mHasAimState)
+            mHandAimPose = aimJoint.pose;
+    }
 
```

Comment:
Alternate approach for the mHandAimPose assignment: 

```If (mHasAimState) mHandAimPose = mSupportsFBHandStrakingAim ? aimState.aimPose : aimJoint.pose;```

Reply:
You cannot do that because aimJoint is only defined in the else branch